uniprot-search/src/main/java/org/uniprot/store/search/domain2/UniProtSearchFields.java

```diff
@@ -1,42 +1,65 @@
 package org.uniprot.store.search.domain2;
 
-import org.uniprot.core.cv.xdb.UniProtXDbTypes;
-import org.uniprot.store.search.domain2.impl.SearchFieldImpl;
-
 import java.util.List;
+import java.util.Objects;
 import java.util.Set;
 import java.util.stream.Collectors;
 
+import org.uniprot.core.cv.xdb.UniProtXDbTypes;
+import org.uniprot.store.search.domain2.impl.SearchFieldImpl;
+
 /**
- * Represents all accessible UniProtKB search fields, and provides access to them via the {@link
- * SearchFields} contract.
+ * Represents all accessible search fields for UniProt domains (e.g., UniProtKB, UniParc, UniRef),
+ * and provides access to them via the {@link SearchFields} contract.
  *
- * <p>Created 12/11/2019
+ * <p>Created 17/01/2020
  *
  * @author Edd
  */
-public enum UniProtKBSearchFields implements SearchFields {
-    INSTANCE;
+public enum UniProtSearchFields implements SearchFields {
+    CROSSREF("crossref/search-fields.json"),
+    DISEASE("disease/search-fields.json"),
+    GENECENTRIC("gene-centric/search-fields.json"),
+    KEYWORD("keyword/search-fields.json"),
+    LITERATURE("literature/search-fields.json"),
+    PROTEOME("proteome/search-fields.json"),
+    SUBCELL("subcell-location/search-fields.json"),
+    TAXONOMY("taxonomy/search-fields.json"),
+    UNIPARC("uniparc/search-fields.json"),
+    UNIPROTKB("uniprot/search-fields.json"),
+    UNIREF("uniref/search-fields.json");
 
-    private static final String FILENAME = "uniprot/search-fields.json";
-    static final String XREF_COUNT_PREFIX = "xref_count_";
-    private UniProtKBSearchFieldsLoader searchFieldsLoader;
+    private String configPath;
+    private SearchFieldsLoader searchFieldsLoader;
 
-    UniProtKBSearchFields() {
-        searchFieldsLoader = new UniProtKBSearchFieldsLoader(FILENAME);
+    UniProtSearchFields(String configPath) {
+        this.configPath = configPath;
     }
 
     @Override
     public Set<SearchField> getSearchFields() {
+        checkInitialised();
         return searchFieldsLoader.getSearchFields();
     }
 
     @Override
     public Set<SearchField> getSortFields() {
+        checkInitialised();
         return searchFieldsLoader.getSortFields();
     }
 
+    private void checkInitialised() {
+        if (Objects.isNull(searchFieldsLoader)) {
+            if (configPath.startsWith("uniprot")) {
+                searchFieldsLoader = new UniProtKBSearchFieldsLoader(configPath);
+            } else {
+                searchFieldsLoader = new SearchFieldsLoader(configPath);
+            }
+        }
+    }
+
     private static class UniProtKBSearchFieldsLoader extends SearchFieldsLoader {
```

Comment:
We should take out UniProtKBSearchFieldsLoader from here and make it a class.

Reply:
done